### ISSUE TITLE
Color problem

### ISSUE THREAD

**Issue by YZBruh** [TrebleDroid/treble_experimentations#91]
The colors on the screen constantly (especially when I am in contact with the screen) become faded and colorful, as if a flash explodes. It does this very quickly. The color is also faded in the screenshots I took when the screen was in that state. And it reduces performance. I will leave a video to explain better.

As I said above, the colors fade and flash rapidly with each contact with the screen. And sometimes it just stays that way.

I want to get rid of the problem...

https://github.com/TrebleDroid/treble_experimentations/assets/131263805/966cc9d0-dbb7-49c9-8866-14aff776736b

The screenshot I'm talking about (don't care which app it is xd):
![Screenshot_20240106-135837_Bugjaeger Premium.png](https://github.com/TrebleDroid/treble_experimentations/assets/131263805/0580811f-5b59-4815-bfad-fdab512f4e11)

 - Device: Alcatel 3T 10 (8088X model. tb8765ap1_bsp)
 - Build Flavor: system-arm32-binder-64-ab-vanilla.img.xz
 - Version: Android 14

**Comment by phhusson** [comment]
Notes: https://www.gsmarena.com/alcatel_3t_10-9601.php / launched with Android 8.1 / SoC is MT8765WB with Imagination PowerVR GPU

**Comment by phhusson** [comment]
Send the result of `adb shell dumpsys SurfaceFlinger` on Settings screen before the screen turns white, then while it is white.

**Comment by YZBruh** [comment]
> Notes: https://www.gsmarena.com/alcatel_3t_10-9601.php / launched with Android 8.1 / SoC is MT8765WB with Imagination PowerVR GPU

The device comes with Android 9 :/

**Comment by YZBruh** [comment]
> Send the result of `adb shell dumpsys SurfaceFlinger` on Settings screen before the screen turns white, then while it is white.

Ok. I'm doing it now

**Comment by YZBruh** [comment]
@phhusson The printouts are too long. So I saved it to a file. There are outputs that are in white color from the first to the 2136th line. After black.

[dumpsys.log](https://github.com/TrebleDroid/treble_experimentations/files/13854917/dumpsys.log)

**Comment by phhusson** [comment]
What happens if you tick `Disable HW overlay` ?

**Comment by YZBruh** [comment]
> What happens if you tick `Disable HW overlay` ?

The screen remains completely white 🤮

**Comment by ArsBinarii** [comment]
Same issue here:
Hardware:

Oukitel k12
CPU MTK: 6765
Android 9 at launch
integrated graphics card is PowerVR GE8320

Symptom:
1) by default image has washed out colors, problem identified as in some cases (scroll/animation..etc real colors are visible)
equivalent to a white overlay with alpha set to ~20%.
2) it seems that the clear screen (where the "overlay" is not present) animate and move faster
3) android startup logo flashing

Disable HW overlay - permanent washed out colors, during animations and transitions the "real color screen" is no longer visible

What I tried:
Disable HW Overlay on/off
ForceFPS from don't force to 1080x2340@60.330006
Mediatek GED KPI Support on/off

[SurfaceFlinger_MTKMT6765_DisableHwO_ON.txt](https://github.com/TrebleDroid/treble_experimentations/files/13983775/SurfaceFlinger_MTKMT6765_DisableHwO_ON.txt)
[SurfaceFlinger_MTKMT6765_DisableHwO_OFF.txt](https://github.com/TrebleDroid/treble_experimentations/files/13983776/SurfaceFlinger_MTKMT6765_DisableHwO_OFF.txt)

**Comment by phhusson** [comment]
Can you also test on Android 10, 11 and 12? (from https://github.com/phhusson/treble_experimentations/releases/ )

**Comment by ArsBinarii** [comment]
yeah, yeah :)) I did 12 - [AOSP 12.1 v416](https://github.com/phhusson/treble_experimentations/releases/tag/v416) and the "flickering" was not there, I am familiar with your work :).

do you need 11 and 10 ?

**Comment by phhusson** [comment]
Nope, knowing that 12 is ok is enough, thanks. Just to confirm, 12 is okay even if you tick "disable gpu overlay"?

**Comment by ArsBinarii** [comment]
correct, Disable HW Overlay on/off 12, keeps the same colors and no issues are to be seen. seems the problem started in with 13.

**Comment by ArsBinarii** [comment]
if I have some more free time later today I will try all 13 builds below 41 maybe I can identify the one that introduced the issue

**Comment by phhusson** [comment]
Could you get me the `dumpsys SurfaceFlinger` with "disable hw overlay" enabled on Android 12?

**Comment by ArsBinarii** [comment]
> Could you get me the `dumpsys SurfaceFlinger` with "disable hw overlay" enabled on Android 12?

[SurfaceFlinger_A12_v416_MTKMT6765_DisableHwO_ON_postReboot.txt](https://github.com/TrebleDroid/treble_experimentations/files/13988894/SurfaceFlinger_A12_v416_MTKMT6765_DisableHwO_ON_postReboot.txt)
[SurfaceFlinger_A12_v416_MTKMT6765_DisableHwO_ON.txt](https://github.com/TrebleDroid/treble_experimentations/files/13988895/SurfaceFlinger_A12_v416_MTKMT6765_DisableHwO_ON.txt)
[SurfaceFlinger_A12_v416_MTKMT6765_DisableHwO_OFF.txt](https://github.com/TrebleDroid/treble_experimentations/files/13988896/SurfaceFlinger_A12_v416_MTKMT6765_DisableHwO_OFF.txt)

**Comment by ArsBinarii** [comment]
@phhusson hey, been thinking about this for a while now, I think the deep and root of the problem is been in the system before A12, the current issue is that now visible since A13. As you can see, this failure would now be visible under A12 since 
disable hw overlay on/off the image do not change.

My hunch is that there is an integration driver problem, with the video driver, while using Pixel Experience A12 (yet, just. confirmed on your A12 build) I remember seeing tearing/strange artifacting while using, the most common point would be the PIN screen while pushing the buttons.

<img width="565" alt="image" src="https://github.com/TrebleDroid/treble_experimentations/assets/6293391/a74ffa44-580e-47b4-8105-5c0a3096f57c">
<img width="581" alt="image" src="https://github.com/TrebleDroid/treble_experimentations/assets/6293391/45c3c719-1061-4501-be95-8dfbf06fe83d">

**Comment by ArsBinarii** [comment]
tested [AOSP 11.0 v313](https://github.com/phhusson/treble_experimentations/releases/tag/v313) no screen tearing or artifacts, but animations as A12, A13, A14 seem chunky (A14 maybe seems a but better) [I usually disable animations via Dev tools].
A11 - tried disable hw overlay on/off [Dev tools]- no white overlay seen, no tearing/artifacting

[SurfaceFlinger_A11_v313_MTKMT6765_DisableHwO_OFF.txt](https://github.com/TrebleDroid/treble_experimentations/files/13999775/SurfaceFlinger_A11_v313_MTKMT6765_DisableHwO_OFF.txt)
[SurfaceFlinger_A11_v313_MTKMT6765_DisableHwO_ON.txt](https://github.com/TrebleDroid/treble_experimentations/files/13999776/SurfaceFlinger_A11_v313_MTKMT6765_DisableHwO_ON.txt)

SurfaceFlinger_A11_v313_MTKMT6765_DisableHwO_ON_postReboot.txt - could not be provided, upon reboot A11 is stuck at animated boot logo and will not load the system

**Comment by mohammad-kazemzadeh** [comment]
@ArsBinarii 

I can confirm animations ARE chunky on A12 - A14 along with scroll lags
even disabling hw overlaying causes the scrolls to be more laggy
I'm testing on Honor 7x (kirin 659)

on A14 animations are better than before but not the way they're supposed to feel

**Comment by YZBruh** [comment]
It was great to find more people... Everyone is experiencing these problems above A12, right?

**Comment by YZBruh** [comment]
It was great to find more people... Everyone is experiencing these problems above A12, right?

**Comment by mohammad-kazemzadeh** [comment]
@YZBruh with high probability YES
@phhusson 
I have not tested A11, But going to test it as soon as possible
because everything besides graphics and animations are great

my phone (Honor 7x) is laggy with chunky animations and scrolling 
but apps seem to launch quite faster than the stock ROM
so I guess the problem is related to graphics of these GSIs 

I really appreciate the hard work & dedication 

Any fix for animations & graphic related issues will be greatly appreciated 

My device: 
Honor 7x
CPU: Kirin 659
GPU: Mali-T830 MP2

**Comment by ArsBinarii** [comment]
kindly open a new ticket for your issue, we are debugging something else here, if fixing this will fix the animations that is a different story

**Comment by YZBruh** [comment]
@phhusson Is there any solution to solve the problem? The problem still exists in the latest android 14 AOSP build. At least it booted... (It wasn't booting before).

**Comment by YZBruh** [comment]
> Just so everybody understands, from my experience, this is a very hard issues to debug and fix and might take a long time to fix, or might even not get fixed at all since this is not occurring on all devices, or due to closed envs that can't be debugged like drivers, implementations, ...etc.
> 
> Just be patient.

Okay. Thanks.

**Comment by YZBruh** [comment]
So is it possible to keep it in a certain mode?

**Comment by YZBruh** [comment]
Then this is impossible... There is no code available. I have no idea...
